lib/propmangle.js

```diff
@@ -116,13 +131,34 @@ function mangle_properties(ast, options) {
         }
         else if (node instanceof AST_Dot) {
             add(node.property);
+            
+            // if the left side of the dot is a global alias (e.g. window.foo), and the left side
+            // does not refer to a local variable, add it to a list of global definitions.
+            if (mangle_globals &&
+                node.expression instanceof AST_SymbolRef &&
+                node.expression.global() &&
+                is_global_alias(node.expression.name) &&
+                !is_global_alias(node.property))
+            {
```

Comment:
Uglify code style puts the appends `{` to the previous line while leaving indenting as you have it. Yes it's unusual.